test/on_error_test.rb

```diff
@@ -0,0 +1,46 @@
+require 'test_helper'
+
+# The on_error call is tested herer.
```

Comment:
I think these tests can just be moved to `options_test` since they all occur on the `Options` class, then no need for this file

Reply:
Can od that. No problem ;-)